agent/exec/controller_test.go

```diff
@@ -21,23 +21,24 @@ func TestResolve(t *testing.T) {
 		ctx      = context.Background()
 		executor = &mockExecutor{}
 		task     = newTestTask(t, api.TaskStateAssigned, api.TaskStateRunning)
+		secrets  map[string]*api.Secret
 	)
 
-	_, status, err := Resolve(ctx, task, executor)
+	_, status, err := Resolve(ctx, task, secrets, executor)
```

Comment:
This passes in a nil map. Is that the intent? If so, how about `Resolve(ctx, task, nil, executor)`? Same for other calls to `Resolve` below.